atlasdb-commons/src/test/java/com/palantir/common/collect/IterableViewTest.java

```diff
@@ -0,0 +1,114 @@
+        int characteristics = spliterator.characteristics();
+        assertThat(characteristics).isNotEqualTo(0);
+        assertThat(characteristics & Spliterator.CONCURRENT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.DISTINCT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.IMMUTABLE).isEqualTo(Spliterator.IMMUTABLE);
+        assertThat(characteristics & Spliterator.NONNULL).isEqualTo(Spliterator.NONNULL);
+        assertThat(characteristics & Spliterator.SIZED).isEqualTo(Spliterator.SIZED);
+        assertThat(characteristics & Spliterator.SORTED).isEqualTo(0);
+        assertThat(characteristics & Spliterator.SUBSIZED).isEqualTo(Spliterator.SUBSIZED);
+    }
+
+    @Test
+    public void listPartitionSpliterator() {
+        IterableView<List<String>> view = IterableView.of(ImmutableList.of("a", "b", "c")).partition(1);
+        Spliterator<List<String>> spliterator = view.spliterator();
+        assertThat(spliterator.estimateSize()).isEqualTo(3);
+        int characteristics = spliterator.characteristics();
+        assertThat(characteristics).isNotEqualTo(0);
+        assertThat(characteristics & Spliterator.CONCURRENT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.DISTINCT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.IMMUTABLE).isEqualTo(0);
+        assertThat(characteristics & Spliterator.NONNULL).isEqualTo(0);
+        assertThat(characteristics & Spliterator.SIZED).isEqualTo(Spliterator.SIZED);
+        assertThat(characteristics & Spliterator.SORTED).isEqualTo(0);
+        assertThat(characteristics & Spliterator.SUBSIZED).isEqualTo(Spliterator.SUBSIZED);
+    }
+
+    @Test
+    public void listTransformSpliterator() {
+        IterableView<String> view = IterableView.of(ImmutableList.of("a", "b", "c")).transform(String::toUpperCase);
+        Spliterator<String> spliterator = view.spliterator();
+        assertThat(spliterator.estimateSize()).isEqualTo(3);
+        int characteristics = spliterator.characteristics();
+        assertThat(characteristics).isNotEqualTo(0);
+        assertThat(characteristics & Spliterator.CONCURRENT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.DISTINCT).isEqualTo(0);
+        assertThat(characteristics & Spliterator.IMMUTABLE).isEqualTo(0);
+        assertThat(characteristics & Spliterator.NONNULL).isEqualTo(0);
+        assertThat(characteristics & Spliterator.SIZED).isEqualTo(Spliterator.SIZED);
+        assertThat(characteristics & Spliterator.SORTED).isEqualTo(0);
+        assertThat(characteristics & Spliterator.SUBSIZED).isEqualTo(Spliterator.SUBSIZED);
+    }
+
+    @Test
+    public void setSpliterator() {
+        Spliterator<String> spliterator = IterableView.of(ImmutableSortedSet.of("a", "b")).spliterator();
+        assertThat(spliterator.estimateSize()).isEqualTo(2);
+        int characteristics = spliterator.characteristics();
+        assertThat(characteristics).isNotEqualTo(0);
```

Comment:
would be tempted to remove these isNotZero style asserts; they're redundant given the below constraints

Reply:
actually left them in as the failure message if this assumption fails is more useful than the true/false of characteristic check.

Reply:
fair